src/bosh_azure_cpi/lib/cloud/azure/vm_manager.rb

```diff
@@ -4,33 +4,105 @@ module Bosh::AzureCloud
   class VMManager
     include Helpers
 
-    def initialize(azure_properties, registry_endpoint, disk_manager, disk_manager2, azure_client2, storage_account_manager)
+    def initialize(azure_properties, registry_endpoint, disk_manager, disk_manager2, azure_client2, storage_account_manager, stemcell_manager, stemcell_manager2, light_stemcell_manager)
       @azure_properties = azure_properties
       @registry_endpoint = registry_endpoint
       @disk_manager = disk_manager
       @disk_manager2 = disk_manager2
       @azure_client2 = azure_client2
       @storage_account_manager = storage_account_manager
-      @use_managed_disks = @azure_properties['use_managed_disks']
+      @stemcell_manager = stemcell_manager
+      @stemcell_manager2 = stemcell_manager2
+      @light_stemcell_manager = light_stemcell_manager
+      @use_managed_disks = azure_properties['use_managed_disks']
       @logger = Bosh::Clouds::Config.logger
     end
 
-    def create(instance_id, location, stemcell_info, resource_pool, network_configurator, env)
+    def create(instance_id, location, stemcell_id, resource_pool, network_configurator, env)
       # network_configurator contains service principal in azure_properties so we must not log it.
-      @logger.info("create(#{instance_id}, #{location}, #{stemcell_info.inspect}, #{resource_pool}, ..., ...)")
-      resource_group_name = instance_id.resource_group_name()
+      @logger.info("create(#{instance_id}, #{location}, #{stemcell_id}, #{resource_pool}, ..., ...)")
+
       vm_name = instance_id.vm_name()
-      vm_size = resource_pool.fetch('instance_type', nil)
 
       # When both availability_zone and availability_set are specified, raise an error
       cloud_error("Only one of `availability_zone' and `availability_set' is allowed to be configured for the VM but you have configured both.") if !resource_pool['availability_zone'].nil? && !resource_pool['availability_set'].nil?
+      zone = resource_pool.fetch('availability_zone', nil)
+      unless zone.nil?
+        cloud_error("`#{zone}' is not a valid zone. Available zones are: #{AVAILABILITY_ZONES}") unless AVAILABILITY_ZONES.include?(zone.to_s)
+      end
+
+      # async task - prepare stemcell. Time consuming works could be:
+      #   * create usage image
```

Comment:
saw a bunch of relocation of the code.
can we seperate the change in two steps:
1. relocate the code.
2. change the relocated code to use the parallel lib.

merge them together make the code review hard.